drivers/overlay/overlay.go

```diff
@@ -1198,16 +1215,25 @@ func (d *Driver) dir2(id string, useImageStore bool) (string, string, bool) {
 
 	newpath := path.Join(homedir, id)
 
+	allImageStores := d.getAllImageStores()
+
 	if err := fileutils.Exists(newpath); err != nil {
-		for _, p := range d.getAllImageStores() {
+		for _, p := range allImageStores {
 			l := path.Join(p, d.name, id)
 			err = fileutils.Exists(l)
 			if err == nil {
-				return l, homedir, true
+				return l, homedir, false, true
+			}
+		}
+		for _, p := range allImageStores {
+			l := path.Join(p, vfs.Name, "dir", id)
```

Comment:
Hard-coding VFS feels a bit surprising.

If the same layer exists if multiple image stores, this will probably not find them in the same order `store.go` will. Does that matter?

Reply:
it should not really matter, but I'll change the lookup code in overlay to behave in the same way as in `store.go`